dnshostprovider.go

```diff
@@ -18,6 +18,22 @@ type DNSHostProvider struct {
 	lookupHost func(string) ([]string, error) // Override of net.LookupHost, for testing.
 }
 
+func addrsByHostname(server string) ([]string, error) {
```

Comment:
nit: since this is used in the function below, this function would be sorted below it.